src/stan/mcmc/windowed_adaptation.hpp

```diff
@@ -40,17 +40,18 @@ namespace stan {
         }
 
         if (init_buffer + base_window + term_buffer > num_warmup) {
-          writer("WARNING: The initial buffer, adaptation window, "
-                 "and terminal buffer");
-          writer("         overflow the total number of warmup iterations.");
+          writer("WARNING: There aren't enough warmup iterations to fit the");
+          writer("         three stages of adaptation as currently"
+                 + std::string(" configured."));
```

Comment:
The idiomatic style for this leans on C++'s implicit concatenation of string literals:

```
writer("         three stages of adaptation as currently"
       " configured");
```

The two strings get concatenated and stored in the program at compile time rather than relying on a run-time concatenation (which generates more code and takes more time, not that either a concern here).